app/src/main/java/org/mozilla/fenix/settings/TextPercentageSeekBarPreference.kt

```diff
@@ -0,0 +1,484 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings
+
+/*
+ * Copyright 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import android.content.Context
+import android.content.res.TypedArray
+import android.os.Parcel
+import android.os.Parcelable
+import android.util.AttributeSet
+import android.util.Log
+import android.util.TypedValue
+import android.view.KeyEvent
+import android.view.View
+import android.widget.SeekBar
+import android.widget.SeekBar.OnSeekBarChangeListener
+import android.widget.TextView
+import androidx.preference.Preference
+import androidx.preference.PreferenceViewHolder
+import org.mozilla.fenix.R
+
+import java.text.NumberFormat
+
+/**
+ * Preference based on android.preference.SeekBarPreference but uses support preference as a base
+ * . It contains a title and a [SeekBar] and a SeekBar value [TextView] and an Example [TextView].
+ * The actual preference layout is customizable by setting `android:layout` on the
+ * preference widget layout or `seekBarPreferenceStyle` attribute.
+ *
+ *
+ * The [SeekBar] within the preference can be defined adjustable or not by setting `adjustable` attribute.
+ * If adjustable, the preference will be responsive to DPAD left/right keys.
+ * Otherwise, it skips those keys.
+ *
+ *
+ * The [SeekBar] value view can be shown or disabled by setting `showSeekBarValue`
+ * attribute to true or false, respectively.
+ *
+ *
+ * Other [SeekBar] specific attributes (e.g. `title, summary, defaultValue, min,
+ * max`)
+ * can be set directly on the preference widget layout.
+ */
+class TextPercentageSeekBarPreference @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = R.attr.seekBarPreferenceStyle,
+    defStyleRes: Int = 0
+) : Preference(context, attrs, defStyleAttr, defStyleRes) {
+    internal /* synthetic access */ var mSeekBarValue: Int = 0
+    internal /* synthetic access */ var mMin: Int = 0
+    private var mMax: Int = 0
+    private var mSeekBarIncrement: Int = 0
+    internal /* synthetic access */ var mTrackingTouch: Boolean = false
+    internal /* synthetic access */ var mSeekBar: SeekBar? = null
+    private var mSeekBarValueTextView: TextView? = null
+    private var mExampleTextTextView: TextView? = null
+    // Whether the SeekBar should respond to the left/right keys
+    /**
```

Comment:
We probably shouldn't have three sets of KDoc including the getter, setter, and variable separately for each property. I recognize this is all from the original JavaDoc in the AOSP Java code.